src/univariate/locationscale.jl

```diff
@@ -1,129 +1,138 @@
-parameter `σ`.
 
-If `ρ` is a discrete distribution, the probability mass function of
-the transformed distribution is given by
+If `ρ` is a discrete univariate distribution, the probability mass function of the
+transformed distribution is given by
 ```math
 P(X = x) = P\\left(Z = \\frac{x-μ}{σ} \\right).
 ```
-If `ρ` is a continuous distribution, the probability density function of
-the transformed distribution is given by
+
+If `ρ` is a continuous univariate distribution with probability density function `f_Z`,
+the probability density function of the transformed distribution is given by
 ```math
-f(x) = \\frac{1}{σ} ρ \\! \\left( \\frac{x-μ}{σ} \\right).
+f_X(x) = \\frac{1}{|σ|} f_Z\\left( \\frac{x-μ}{σ} \\right).
 ```
 
+We recommend against using the `AffineDistribution` constructor directly. Instead, use
+`+`, `-`, `*`, and `/`. These are optimized for specific distributions and will fall back
+on `AffineDistribution` only when they need to.
+
+Affine transformations of discrete variables are easily affected by rounding errors. If you
+are getting incorrect results, try using exact `Rational` types instead of floats.
+
 ```julia
-LocationScale(μ,σ,ρ) # location-scale transformed distribution
-params(d)            # Get the parameters, i.e. (μ, σ, and the base distribution ρ)
-location(d)          # Get the location parameter
-scale(d)             # Get the scale parameter
+d = σ * ρ + μ       # Create location-scale transformed distribution
+params(d)           # Get the parameters, i.e. (μ, σ, ρ)
 ```
-
-External links
-[Location-Scale family on Wikipedia](https://en.wikipedia.org/wiki/Location%E2%80%93scale_family)
 """
-struct LocationScale{T<:Real, S<:ValueSupport, D<:UnivariateDistribution{S}} <: UnivariateDistribution{S}
+struct AffineDistribution{T<:Real, S<:ValueSupport, D<:UnivariateDistribution{S}} <: UnivariateDistribution{S}
     μ::T
     σ::T
     ρ::D
-    function LocationScale{T,S,D}(μ::T, σ::T, ρ::D; check_args=true) where {T<:Real, S<:ValueSupport, D<:UnivariateDistribution{S}}
-        check_args && @check_args(LocationScale, σ > zero(σ))
+    function AffineDistribution{T,S,D}(μ::T, σ::T, ρ::D; check_args=true) where {T<:Real, S<:ValueSupport, D<:UnivariateDistribution{S}}
+        check_args && @check_args(AffineDistribution, σ > zero(σ))
         new{T, S, D}(μ, σ, ρ)
     end
 end
 
-function LocationScale(μ::T, σ::T, ρ::UnivariateDistribution; check_args=true) where {T<:Real}
+function AffineDistribution(μ::T, σ::T, ρ::UnivariateDistribution; check_args=true) where {T<:Real}
     _T = promote_type(eltype(ρ), T)
     D = typeof(ρ)
     S = value_support(D)
-    return LocationScale{_T,S,D}(_T(μ), _T(σ), ρ; check_args=check_args)
+    return AffineDistribution{_T,S,D}(_T(μ), _T(σ), ρ; check_args=check_args)
 end
 
-LocationScale(μ::Real, σ::Real, ρ::UnivariateDistribution) = LocationScale(promote(μ, σ)..., ρ)
+AffineDistribution(μ::Real, σ::Real, ρ::UnivariateDistribution) = AffineDistribution(promote(μ, σ)..., ρ)
```

Comment:
Maybe this fixes the downstream tests (should be fixed anyway):
```suggestion
function AffineDistribution(μ::Real, σ::Real, ρ::UnivariateDistribution; check_args=true)
    return AffineDistribution(promote(μ, σ)..., ρ; check_args=check_args)
end
```